### PR TITLE
[RLlib] Do-over of release tests in light of rllib_contrib AND new- vs old API stack.

### PR DESCRIPTION
**Pull request by sven1977**
Do-over of release tests in light of rllib_contrib AND new- vs old API stack.

Breakdown:
* PPO now has its first new stack + ConnectorV2 (single-agent) 8GPU + 95 EnvRunner  release test for Atari (Pong (~5min) and Breakout (~30-45min)).
* We no longer run PPO hybrid stack release tests (too much burden to maintain and will be replaced completely by new API stack).
* We no longer run contrib algos release tests (too much burden to maintain).
* Retire IMPALA (tf-only) stress test. The "stress" bit of the original intent behind this test has been moved into the new PPO 8GPUS + 95worker test (see above).
* Retire old multi-gpu tests. All to be replaced by new API stack 2-8 GPUs learning tests (which will become the norm!).

Main TODO (this was never done for the old stack either): Multi-agent release tests on harder benchmarks!

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/rllib_tests/learning_tests/yaml_files/ppo/new_stack/ppo_breakout.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    return FrameStackingLearner(num_frames=4)
+
+
+# Create a custom Atari setup (w/o the usual RLlib-hard-coded framestacking in it).
+# We would like our frame stacking connector to do this job.
+def _env_creator(cfg):
+    return wrap_atari_for_new_api_stack(
+        gym.make("ALE/Breakout-v5", **cfg, **{"render_mode": "rgb_array"}),
+        # Perform through ConnectorV2 API.
+        framestack=None,
```
Comment (simonsays1980) [inline release/rllib_tests/learning_tests/yaml_files/ppo/new_stack/ppo_breakout.py:25]: Is this intended bheavior or only for test? Should users keep using framestack wrappers or use conectors?

**Review by simonsays1980 — APPROVED** [review]
LGTM. Running the new stack with >50 workers :)

File: release/rllib_tests/learning_tests/yaml_files/ppo/new_stack/ppo_breakout.py (hunk 1 above)
Comment (simonsays1980) [reply release/rllib_tests/learning_tests/yaml_files/ppo/new_stack/ppo_breakout.py:25]: Thanks for the clarification. We should later collect all possible speedups we find and remember together and give users a speed-up guide that navigates them a bit through the  labyrinth - especially in the areas where these changes would contradict our design mantras. 

On the side: why only after sampling and before training? does framestack not usually require a different encoder?

**Review by kouroshHakha — COMMENTED** [review]
> We no longer run contrib algos release tests (too much burden to maintain).

Why? we should only run them where there is a change in those sub-folders. That's how it should have been setup.

**Pull request merged**